public/app/features/transformers/extractFields/fieldExtractors.ts

```diff
@@ -133,7 +133,26 @@ const extLabels: FieldExtractor = {
   getParser: (options) => parseKeyValuePairs,
 };
 
-const fmts = [extJSON, extLabels, extRegExp];
+const extCSV: FieldExtractor = {
+  id: FieldExtractorID.CSV,
+  name: 'Delimiter separated values',
+  description: 'Splits at delimited values, such as commas',
+  getParser: ({ delimiter = ',' }) => {
+    return (raw: string) => {
+      // Try to split delimited values
+      const parts = raw.split(delimiter).map((v) => v.trim());
+      if (parts.length < 1) {
+        return undefined;
+      }
+      return parts.reduce((acc: Record<string, boolean>, part: string) => {
+        acc[part] = true;
+        return acc;
+      }, {});
+    };
+  },
+};
```

Comment:
maybe something at avoids function loops and extra array allocations:

```suggestion
  getParser: ({ delimiter = ',' }) => {
    const splitRegExp = new RegExp(`\\s*${escapeStringForRegex(delimiter)}\\s*`, 'g');

    return (raw: string) => {
      const parts = raw.trim().split(splitRegExp);

      if (parts.length === 0) {
        return;
      }

      const acc: Record<string, boolean> = {};
      for (let part of parts) {
        acc[part] = true;
      }
      return acc;
    };
  },
```